beacon_node/network/src/sync/block_sidecar_coupling.rs

```diff
@@ -133,6 +186,59 @@ impl<E: EthSpec> RangeBlockComponentsRequest<E> {
         }
     }
 
+    /// Initialize the entries for this component after the column requests have been sent.
+    pub fn initialize_data_columns_from_root_component(
+        &mut self,
+        column_requests: Vec<(DataColumnsByRootRequestId, Vec<ColumnIndex>)>,
+    ) -> Result<(), String> {
+        // Nothing to insert, do not initialize
+        if column_requests.is_empty() {
+            return Ok(());
+        }
+        match &mut self.block_data_request {
+            RangeBlockDataRequest::DataColumnsFromRoot {
+                requests,
+                attempt: _,
+                all_requests_made,
+                column_peers,
+                expected_custody_columns,
+            } => {
+                for (request, peers) in column_requests {
+                    requests.insert(request, ByRangeRequest::Active(request));
+                    column_peers.insert(request, peers);
+                }
+
+                if !*all_requests_made {
+                    let mut all_columns_requested = HashSet::new();
+                    for columns in column_peers.values() {
+                        all_columns_requested.extend(columns.iter());
+                    }
+                    *all_requests_made = all_columns_requested == *expected_custody_columns;
+                }
+
+                Ok(())
+            }
+            _ => Err("Invalid initialization".to_string()),
+        }
+    }
+
+    /// This modifies the internal variant to `NoData`.
+    ///
+    /// Once we make the block request for a batch and get responses, it is possible
+    /// that the entire batch contained no blobs based on the values of `expected_kzg_commitments`.
+    ///
+    /// At this point, we do not need to make any requests and the blocks correspond to all the
+    /// available data for this batch. Hence, we indicate here that this component requires no data.
+    pub fn no_columns_for_batch(&mut self) -> Result<(), String> {
```

Comment:
Are all String errors returned in this struct impl internal errors that doesn't need different handling?